awscli/clidocs.py

```diff
@@ -177,6 +178,19 @@ def doc_relateditem(self, help_command, related_item, **kwargs):
         )
         doc.write('\n')
 
+    def _document_enums(self, argument, doc):
+        """Documents top-level parameter enums"""
+        model = getattr(argument, 'argument_model', {})
+        metadata = getattr(model, 'metadata', {})
+        enums = metadata.get('enum', [])
```

Comment:
What is the reason in not using the work you did for this PR: https://github.com/boto/botocore/pull/608/files here? I also noticed the use of enum values in the JSON syntax related methods such as here: https://github.com/aws/aws-cli/blob/develop/awscli/clidocs.py#L43. It would be nice if we used what you added there as well.